### ISSUE TITLE
Hardware: Have a single board computer spare? Will it run PiNode-XMR?

### ISSUE THREAD

**Issue by shermand100** [monero-ecosystem/PiNode-XMR#3]
## I'd like some assistance from the community!

With the self installer scripts releasing us from the constraints of the Raspberry Pi platform I require testers with access to single board computers (SBC). It would be rather wasteful (and expensive) for me to buy one of every SBC out there so I'm asking you to dig though your spare project box's to see what you can put to use.

The list below is not exhaustive. If you own something else that's not on the list, that's fine, give it a go, it can be added. Use the list as a guide for the info we need. If you use a non-standard disk image can you link where others can find it, however standard images should be used where possible as the point of this open source installer is that it can be trusted.

Leave your findings as a comment at the bottom.

**My Expectations:**  The purpose of this is to see what variations (if any) affect the PiNode-XMR installer or how it functions once complete. I believe that on each device the node will install and be functional if using Debian Buster, but variations in hardware such as CPU temperature sensors may become inaccurate. Details like this will be logged in the notes column, or a link to their own issue as they arise.

| Device          | Base disk Image | Tested?|   Status  | Build time        | Notes |
| :--------------:| :-------------: |:------:| :------:  |:-----------------:|:-----:|
| Pine A64-LTS    | [buster_current](https://dl.armbian.com/pine64/)  | PENDING| ❓        |❓                 |     |
| Rock 64 | [buster_current](https://dl.armbian.com/rock64/)  | PENDING| ❓        |❓                 |     |
| Le Potato| [buster_current_minimal](https://dl.armbian.com/lepotato/)  | PENDING| ❓        |❓                 |     |
| Banana Pi M64| [buster_current_minimal](https://dl.armbian.com/bananapim64/)  | PENDING| ❓        |❓                 |     |
| Odroid C2 | [buster_current_minimal](https://dl.armbian.com/odroidc2/)  | PENDING| ❓        |❓                 |     |
| Rock Pi 4B| [buster_current](https://dl.armbian.com/rockpi-4b/)  | PENDING| ❓        |❓                 |     |
| NanoPC-T3 Plus| [buster](https://dl.armbian.com/nanopct3plus/)  | PENDING| ❓        |❓                 | Legacy base images listed    |
|Asus Tinkerboard| [buster_current_minimal](https://dl.armbian.com/tinkerboard/)  | PENDING| ❓        |❓                 |     |

### Confirmed

| Device          | Base disk Image | Tested?|   Status  | Build time        | Notes |
| :--------------:| :-------------: |:------:| :------:  |:-----------------:|:-----:|
| Raspberry Pi 3b+| [Raspbian Buster lite 2020-02-13](https://www.raspberrypi.org/downloads/raspbian/)   | ✅ | ✅ |Approx 8hrs|        |
| Raspberry Pi 4  | [Raspbian Buster lite 2020-02-13](https://www.raspberrypi.org/downloads/raspbian/)   | ✅ | ✅ |Approx 4hrs|         |
| Odroid XU4      | [Buster_current_5.4.28_minimal](https://dl.armbian.com/odroidxu4/) | ✅ | ✅ |Approx 4hrs|    |
| Orange Pi Plus 2E| [buster_current](https://dl.armbian.com/orangepiplus2e/)  | ✅  | ✅ | Approx 4-6 hours  | Community tested by @dalecooper  |

**Comment by shermand100** [comment]
Parent post edited. Added Rock64 to confirmed.

| Rock 64 | [buster_current](https://dl.armbian.com/rock64/)  | ✅| ✅|Approx 4 hrs | Requires 4GB RAM (can be supplemented by swap-file)|

Built on 2GB model and required an additional 2GB swapfile. It seems the 64-bit builds are more RAM heavy at Monero build level than a 32-bit build.

**Comment by lobster-kerouac** [comment]
Hello! I just setup PiNode-XMR on a RockPi 4B. I'm planning on writing up a detailed description soon but for now the main points are:

My setup is a RockPi 4B with OS on an eMMC and WD Black 500Gb PCIe NVMe for blockchain storage. Here are some setup notes:

1. The rc-local service was not enabled by default so I had to manually run boot.sh after the reboot steps in the installation process. (I've since enabled rc-local so that the node can self restart after a shutdown).

2. Related, raspi-config doesn't exist (no surprise there), so boot.sh crashed with BOOT_STATUS=1. No problem, I manually changed BOOT_STATUS=2 and continued from there.

3. Building monero took ~3 hours. Nice!

4. I initially was syncing the blockchain over wireless and it was going sloooow and often crashed. When I plugged in the ethernet cord, though, it synced the entire blockchain (block ~2200000) in ~10 hours. Hardware AES for the win!

5. I have some anecdotal evidence that the UDF filesystem was another bottleneck during the initial sync. I eventually formatted the drive to ext4 manually and noticed a speed improvement. This is while I was still struggling with the network stuff, though, so I can't say for sure that ext4 made the difference.

I was thinking of forking a RockPi specific branch to address some of the setup bumps I ran into. Would this be useful? Thanks again for all your hard work!

**Comment by shermand100** [comment]
@lobster-kerouac Sorry I didn't see your post sooner. I don't know why it didn't notify me.

So yeah that rc.local execution was 644 not 755 so that's been updated https://github.com/monero-ecosystem/PiNode-XMR/commit/a1fae6ba9915d78bc2d8469926d55af57a3bff21

The UDF filesystem is a decision that keeps coming back to me and I wonder if it was the right thing. The idea was to allow someone to take the USB drive from PiNodeXMR onto any system (Mac, Linux or Windows) and copy a trusted blockchain. I now wonder if a standard filesystem such as ext4 would be better and then advise on a LAN FTP transfer of the blockchain. I'd go with whichever is easiest for a beginner. A intermediate user could format the drive to whatever they want and mount it, PiNodeXMR wouldn't know/notice.

**Comment by PSLLSP** [comment]
Odroid C4?
Odroid HC4?
Odroid M1?

These are 64-bit. HC4 is similar to C4 but has 2 SATA ports, M1 has M.2 slot and SATA port, I cannot see them in the table but I assume these could be good nodes.

**Comment by shermand100** [comment]
> Odroid C4? Odroid HC4? Odroid M1?
> 
> These are 64-bit. HC4 is similar to C4 but has 2 SATA ports, M1 has M.2 slot and SATA port, I cannot see them in the table but I assume these could be good nodes.

Yes, they can make good nodes as the base images supported by our installer scripts can be found easily...

https://www.armbian.com/download/?tx_maker=hardkernel

https://www.armbian.com/odroid-hc4/

What makes my hardware decisions these days is usually cost efficiency. The latest model (I think) HC4 was released late 2020, so for the unit price in relation to its age should be considered but also the additional cost for accessories. For that reason I prefer the Rock Pi 4 for cheap accessories ( case, emmc module - faster than microSD and longer life , heatsink, power supply etc).

Of course if you already own one then PiNodeXMR would be a great use for it.

**Comment by yijiechoo16163** [comment]
Hi @shermand100 

I tested PiNodeXMR on Raspberry Pi 5 (8 GB RAM model). 
The OS used is Ubuntu 23.10 (GNU/Linux 6.5.0-1011-raspy aarch64)
I am using MicroSD as OS and external SSD as blockchain storage. 
Its build time is 1 hour and 30 minutes.

**Comment by shermand100** [comment]
Thanks for the update @yijiechoo16163 I've added the Pi5 to the table above and also the hardware wiki page. Thanks for testing that. It's a surprisingly fast build time!

**Comment by loopoor** [comment]
Hi , I have installed pinodexmr on ubuntu 22.04 on a atomicpi and runs well
greatings

**Comment by hans-nl** [comment]
Yesterday I installed PiNodeXMR on an OrangePi 5 with 8GB RAM, a 500GB M.2 NVME disk and with a 1Gb ftth internet connection.
The installed OS is Armbian 25.5.1 bookworm minimal (Debian 12) with kernel 6.1.115. Both the OS, PiNodeXMR and the blockchain are on the NVME disk.
Installation/compile took 1h25', downloading the blockchain was between 6 and 15 hours. It's running fine now.
